data_store/aws.py

```diff
@@ -86,11 +86,17 @@ def get_failed_file_key(failure_uuid: UUID) -> str:
     return file_key
 
 
-def get_file_metadata(bucket_name: str, file_key: str) -> Dict[str, str]:
+def get_file_metadata(
+    bucket_name: str,
+    file_key: str,
+    return_full_head_data: bool = False,
```

Comment:
I don't really love parameters that just toggle the return type - that feels more like you have two separate functions and makes it a bit harder to reason about.

This function is specifically about getting the S3 object Metadata tags - not arbitrary metadata. I think this crosses over a bit with what `get_find_download_file_metadata` does - can we refactor these to get one consistent sensible thing? Something like `get_s3_object_attributes` (finger-in-the-air name)?

This change also breaks the typing on this function's return signature (which I've now gone down a bit of a rabbit-hole on - looks like we need to be installing `boto3-stubs[essential]`, not `types-boto3` 🤦)

Reply:
I changed this to be.. `get_file_header`. 

You'll let me know what you think of that on the new version. It returns the whole file header info. If people want metadata they will do `file_header["Metadata"]`, accepts a flag `formatted:bool`, if they want the human readable formatting.